src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +109,63 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+/**
+ * Recursively searches directory and all nested directories until a TypeScript file is found
+ * @param dir Directory to search
+ * @returns true if at least one TypeScript file exists
+ */
+function hasTypeScriptFilesRecursive(dir:string): boolean {
+    const files = readdirSync(dir)
+
+    for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        
+        if (file.substr(-3) === '.ts'){
+            return true
+        }
+    
+        if(statSync(path.join(dir, file)).isDirectory()){
+            if(hasTypeScriptFilesRecursive(path.join(dir, file))) {
+                return true
+            }
+        }
+    }        
+    return false
+}
+
+/**
+ * For non-template debug configs (target = code), compile the project
+ * using a temporary default tsconfig.json file.
+ */
+async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
+    // Compile step for TS projects from source code
+    if (config.invokeTarget.target === 'code') {
+        const samBuildOutputAppRoot = path.join(config.baseBuildDir!, 'output', 'awsToolkitSamLocalResource')
+        const tsconfigPath = path.join(samBuildOutputAppRoot, 'tsconfig.json')
+        if ((await readdir(config.codeRoot)).includes('tsconfig.json') || hasTypeScriptFilesRecursive(config.codeRoot)) {
+            const defaultTsconfig = `{
+                "compilerOptions": {
+                    "target": "es2017",
+                    "module": "commonjs",
+                    "typeRoots": [
+                    "node_modules/@types"
+                    ],                       
+                    "types": [
+                    "node"
+                    ],
+                    "inlineSourceMap": true,
+                    "esModuleInterop": true,
+                    "skipLibCheck": true
+                }
+                }`
```

Comment:
JSON can be defined in code, then stringified. That adds some safety and is also easier to read for humans.

```
const defaultTsconfig = {
    "compilerOptions": {
      ...
    }
}
writeFileSync(..., JSON.stringify(defaultTsconfig))
```

Reply:
Fixed